openedx/core/djangoapps/safe_sessions/middleware.py

```diff
@@ -272,6 +272,8 @@ def sign(self, user_id):
             salt=self.key_salt, algorithm=settings.DEFAULT_HASHING_ALGORITHM
         ).sign_object(data_to_sign, serializer=signing.JSONSerializer, compress=False)
 
+        set_custom_attribute('SafeCookieData.verify.alg', settings.DEFAULT_HASHING_ALGORITHM)
```

Comment:
You only need to set `settings.DEFAULT_HASHING_ALGORITHM` in one place. I'd recommend using `DEFAULT_HASHING_ALGORITHM` as the attribute name, and doing it near the top of the most relevant middleware.